src/Npgsql/MultiplexingDataSource.cs

```diff
@@ -56,13 +58,12 @@ sealed class MultiplexingDataSource : PoolingDataSource
         _connectionLogger = dataSourceConfig.LoggingConfiguration.ConnectionLogger;
         _commandLogger = dataSourceConfig.LoggingConfiguration.CommandLogger;
 
-        // TODO: Think about cleanup for this, e.g. completing the channel at application shutdown and/or
-        // pool clearing
-        _ = Task.Run(MultiplexingWriteLoop, CancellationToken.None)
+        _multiplexWriteLoop = Task.Run(MultiplexingWriteLoop, CancellationToken.None)
             .ContinueWith(t =>
             {
                 // Note that we *must* observe the exception if the task is faulted.
-                _connectionLogger.LogError(t.Exception, "Exception in multiplexing write loop, this is an Npgsql bug, please file an issue.");
+                if (t.Exception?.InnerException is not ChannelClosedException)
```

Comment:
Why not just swallow this exception inside MultiplexingWriteLoop below, and complete normally (return without an exception)?

In fact, it seems that MultiplexingWriteLoop is written so as to prevent any exceptions from escaping... We could even choose to remove the ContinueWith here - or leave it just in case we have a coding bug and an exception does escape.

Reply:
Just convenience really, as we were already handling here every exception, and I didn't want to add another try/catch below.

Reply:
Ah wait, we can probably just do return there...

Reply:
Yep, that's what I had in mind.